### DISCUSSION TITLE
Lesson 4 : can't see transactions on Ganache

### DISCUSSION THREAD

**Discussion by risktakereth** [smartcontractkit/full-blockchain-solidity-course-py#608] [Q&A]
Hello guys, I have an issue. At 4:07:50 in the video, when you try to make your first transaction on Ganache, it just doesnt appear on Ganache. Does someone has the solution ?

Here is my code :

```Python
from solcx import compile_standard, install_solc
import json
from web3 import Web3


with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

# Compile our Solidityyyy

install_solc("0.6.0")

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            },
        },
    },
    solc_version="0.6.0",
)

with open("compile_code.json", "w") as file:
    json.dump(compiled_sol, file)

# get the bytecode motherfucker

bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
    "bytecode"
]["object"]

# get this fuckin abi

abi = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["abi"]

# let's connect to this wtf gannache

w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:7545"))
chain_id = 1337
my_adress = "0x2953809b1392b61204C52B805E916c156DFCf912"
private_key = "0x58b73f588ed5737218eaa0ac120f737cf37d94e50fcce7533e454741c000227b"

# create the lol contract in python

SimpleStorage = w3.eth.contract(abi=abi, bytecode=bytecode)

# get lastest transac bro
nonce = w3.eth.getTransactionCount(my_adress)

# 1. Build a Tx
# 2. Sign a Tx
# 3. Send a Tx

transaction = SimpleStorage.constructor().buildTransaction(
    {
        "gasPrice": w3.eth.gas_price,
        "chainId": chain_id,
        "from": my_adress,
        "nonce": nonce,
    },
)

signed_txn = w3.eth.account.sign_transaction(transaction, private_key=private_key)

# send this signed transaction

tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)

print(tx_hash)
```

I successfully print tx-hash on VSC, so maybe there is no problem in the code. The txn just doesn't appear in Ganache

**Comment by risktakereth** [comment]
Btw, Remix was very nice, but this VSC aprt is getting my angry, it feels like I'm just walking in the mud

**Comment by risktakereth** [comment]
![ok](https://user-images.githubusercontent.com/96295854/146994113-e3f3c4a3-a5fe-43a1-89d5-85b794c349b5.png)
Terminal gives this. Even when I try with this code (https://github.com/PatrickAlphaC/web3_py_simple_storage/blob/main/deploy.py), it doesn't work. Does anyone have an idea ?

**Comment by risktakereth** [comment]
Hm, I just did it again and again until it works...
I honestly can't explain. If anyone has a clear idea of what's going one, feel free X)

**Reply by cromewar** [reply]
Welcome to blockchain development, well not really, but python web3 is known to be a little bit fussy something, anyways please take en consideration WSL as I mentioned bellow, is not perfect, but I will be better.

**Comment by cromewar** [comment]
Hello @GithubSlayer I have never seen that error before, is quite specific but I'm going to suggest you some tips:

1. Use a python virtualenv (you can install it using pip)
2. Don't use windows, ok I'm kidding but actually there is something called WSL (windows subsystem for linux) which is going yo make your life easier, believe you'll have a lot more problem with windows on the next course sections, you can refer to the course index "windows support" to learn how to setup windows for smart contract development.

Hope this info might help.

**Reply by risktakereth** [reply]
Thank you for this piece of advice @cromewar 
I'm going to finish Lesson 4, I will sleep a lot, and I will try this WSL thing tomorrow ! X)
